test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1

```diff
@@ -200,4 +205,614 @@ Describe "TabCompletion" -Tags CI {
+            $res.CompletionMatches[0].CompletionText | Should Be "Overridden-TabExpansion-Alias"
+        }
+    }
+
+    Context "No tab completion tests" {
+        BeforeAll {
+            $testCases = @(
+                @{ inputStr = 'function new-' }
+                @{ inputStr = 'filter new-' }
+                @{ inputStr = '@pid.' }
+            )
+        }
+
+        It "Input '<inputStr>' should not complete to anything" -TestCases $testCases {
+            param($inputStr)
+
+            $res = TabExpansion2 -inputScript $inputStr -cursorColumn $inputStr.Length
+            $res.CompletionMatches.Count | Should BeExactly 0
+        }
+    }
+
+    context "Tab completion error tests" {
+        BeforeAll {
+            $ast = {}.Ast;
+            $tokens = [System.Management.Automation.Language.Token[]]@()
+            $testCases = @(
+                @{ inputStr = {[System.Management.Automation.CommandCompletion]::MapStringInputToParsedInput('$pid.', 7)}; expected = "PSArgumentException" }
+                @{ inputStr = {[System.Management.Automation.CommandCompletion]::CompleteInput($null, $null, $null, $null)}; expected = "PSArgumentNullException" }
+                @{ inputStr = {[System.Management.Automation.CommandCompletion]::CompleteInput($ast, $null, $null, $null)}; expected = "PSArgumentNullException" }
+                @{ inputStr = {[System.Management.Automation.CommandCompletion]::CompleteInput($ast, $tokens, $null, $null)}; expected = "PSArgumentNullException" }
+                @{ inputStr = {[System.Management.Automation.CommandCompletion]::CompleteInput('$pid.', 7, $null, $null)}; expected = "PSArgumentException" }
+                @{ inputStr = {[System.Management.Automation.CommandCompletion]::CompleteInput('$pid.', 5, $null, $null)}; expected = "PSArgumentNullException" }
+                @{ inputStr = {[System.Management.Automation.CommandCompletion]::CompleteInput($null, $null, $null, $null, $null)}; expected = "PSArgumentNullException" }
+                @{ inputStr = {[System.Management.Automation.CommandCompletion]::CompleteInput($ast, $null, $null, $null, $null)}; expected = "PSArgumentNullException" }
+                @{ inputStr = {[System.Management.Automation.CommandCompletion]::CompleteInput($ast, $tokens, $null, $null, $null)}; expected = "PSArgumentNullException" }
+                @{ inputStr = {[System.Management.Automation.CommandCompletion]::CompleteInput($ast, $tokens, $ast.Extent.EndScriptPosition, $null, $null)}; expected = "PSArgumentNullException" }
+            )
+        }
+
+        It "Input '<inputStr>' should throw in tab completion" -TestCases $testCases {
+            param($inputStr, $expected)
+            $inputStr | ShouldBeErrorId $expected
+        }
+    }
+
+    Context "DSC tab completion tests" {
+        BeforeAll {
+            $testCases = @(
+                @{ inputStr = 'Configura'; expected = 'Configuration' }
```

Comment:
Need tests against `import-dscresource` and (I think) the imported resource itself